Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>

From b9126620d74aed1a0ddd286e4c70344626dd72c3 Mon Sep 17 00:00:00 2001
From: Vishnu Banavath <vishnu.banavath@arm.com>
Date: Fri, 17 Dec 2021 19:49:02 +0000
Subject: [PATCH 2/3] efi_loader: populate ESRT table if EFI_ESRT config option
 is set

This change is to call efi_esrt_populate function if CONFIG_EFI_ESRT
is set. This will populte esrt table with firmware image info

Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>
---
 lib/efi_loader/efi_capsule.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/efi_loader/efi_capsule.c b/lib/efi_loader/efi_capsule.c
index ec89865c98..00b3b21105 100644
--- a/lib/efi_loader/efi_capsule.c
+++ b/lib/efi_loader/efi_capsule.c
@@ -591,6 +591,13 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 			ret = EFI_SUCCESS;
 		}
 
+		if (IS_ENABLED(CONFIG_EFI_ESRT)) {
+			/* Rebuild the ESRT to reflect any updated FW images. */
+			ret = efi_esrt_populate();
+	               if (ret != EFI_SUCCESS)
+				log_warning("EFI Capsule: failed to update ESRT\n");
+	       }
+
 		goto out;
 
 #endif
-- 
2.25.1

